tensorflow/compiler/mlir/tosa/transforms/legalize_tfl.cc

```diff
@@ -2168,7 +2169,7 @@ LogicalResult ConvertTFLReshapeOp::matchAndRewrite(
   // Extract the dynamically shaped values for each dimension.
   SmallVector<Value> shape_vals;
   shape_vals.reserve(rank);
-  auto shape_ty = dyn_cast<ShapedType>(shape.getType());
+  auto shape_ty = shape.getType().dyn_cast<ShapedType>();
```

Comment:
I think this should be a cast instead of dyn_cast. The problem is in the original code, but if we're changing the line, we can fix this at the same time.